### PR TITLE
misc: rename lighthouse-(core,cli) folders

### PR DESCRIPTION
**Pull request by connorjclark**
successor to #12859

Only thing worth reviewing are the few opinionated changes I made to the docs links. check out branch and do: `git diff origin/master -- '**/*.md'`

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -394,7 +394,7 @@ Lighthouse now offers auditing user flows, scripted with Puppeteer, via the Node
 
 * publish: include the report bundle in npm package ([#13349](https://github.com/GoogleChrome/lighthouse/pull/13349))
 * build: create inline-fs rollup plugin to replace brfs ([#13232](https://github.com/GoogleChrome/lighthouse/pull/13232), [#13240](https://github.com/GoogleChrome/lighthouse/pull/13240), [#13248](https://github.com/GoogleChrome/lighthouse/pull/13248), [#13272](https://github.com/GoogleChrome/lighthouse/pull/13272), [#13275](https://github.com/GoogleChrome/lighthouse/pull/13275), [#13278](https://github.com/GoogleChrome/lighthouse/pull/13278), [#13280](https://github.com/GoogleChrome/lighthouse/pull/13280))
-* build: use rollup to build lighthouse-core bundles ([#12771](https://github.com/GoogleChrome/lighthouse/pull/12771))
```
Comment (adamraine) [inline changelog.md:397]: Changelog should not be updated

**Review by adamraine — CHANGES_REQUESTED** [review]

File: docs/architecture.md [hunk 2]
```diff
@@ -86,11 +86,11 @@ The return value of each audit [takes this shape](https://github.com/GoogleChrom
 
 The `details` object is parsed in report-renderer.js. View other audits for guidance on how to structure `details`.
 
-## Lighthouse-core internal module dependencies
+## core internal module dependencies
```
Comment (adamraine) [inline docs/architecture.md:89]:
```suggestion
## Core internal module dependencies
```

File: docs/hacking-tips.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ Getting errors like these?
 Use [`--trace-warnings`](https://medium.com/@jasnell/introducing-process-warnings-in-node-v6-3096700537ee) to get actual stack traces.
 
 ```sh
-node --trace-warnings lighthouse-cli http://example.com
```
Comment (adamraine) [inline docs/hacking-tips.md:33]: Would it be possible to create a CLI alias just for this, without doing any package.json export BS?

File: docs/user-flows.md [hunk 4]
```diff
@@ -40,7 +40,7 @@ In DevTools, navigation is easy: ensure it's the selected mode and then click _A
 ```js
 import {writeFileSync} from 'fs';
 import puppeteer from 'puppeteer';
-import lighthouse from 'lighthouse/lighthouse-core/fraggle-rock/api.js';
```
Comment (adamraine) [inline docs/user-flows.md:43]: Should we merge the docs change into a staging branch like we did with https://github.com/GoogleChrome/lighthouse/pull/14195?

File: docs/hacking-tips.md (hunk 3 above)
Comment (connorjclark) [reply docs/hacking-tips.md:33]: you can add lighthouse-core/ to your local gitignore and do whatever you want in there

File: docs/user-flows.md (hunk 4 above)
Comment (connorjclark) [reply docs/user-flows.md:43]: no I don't want to do that for this

**Comment by adamraine** [comment]
I saw several usages of these patterns in `report/README.md` blob links. Can you update those links?

File: docs/user-flows.md (hunk 4 above)
Comment (adamraine) [reply docs/user-flows.md:43]: Why not?

Comment (adamraine) [reply docs/user-flows.md:43]: Oh cuz links will be broken all over the place right

**Comment by connorjclark** [comment]
> I saw several usages of these patterns in `report/README.md` blob links. Can you update those links?

I simply kept those referring to the old commits.

**Review by adamraine — APPROVED** [review]

**Review by Moi0 — COMMENTED** [review]
~Chelbi #220260628

**Pull request merged**